### PR TITLE
refactor: First import of Compose icons and assets and migrate doc files to ImageVector usage

### PR DESCRIPTION
**Pull request by GerardPaligot**
## Changes description 🧑‍💻

* First import of Vitamin Compose icons
* First import of Vitamin Compose assets
* Migrate documentations and sample code of components to `ImageVector` usage
* Use `ImageVector` internally in Vitamin components

## Context 🤔

Will allow to submit automatically new icons and assets from Figma

## Checklist ✅

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check compose contract convention. It must follow conventions described [here](/CONVENTIONS.md).
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.
- [x] I have tested on a phone device/emulator.
- [x] I have tested on a tablet device/emulator.
- [x] I have tested on a large screen device/emulator.
- [x] If it includes design changes, please ask for a review `design-system-core-team-design` GitHub team.

### CODE REVIEW

**Comment by dktbitrise** [comment]
### You can download the apk here :

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/a0e8a49f-8ec0-4ab0-9c2f-fd74f7ab6360/artifact/77ae3f3a856d61e6/p/6d646e21d49154975399f5539f48a1f2

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/cc9c6051-8299-43ba-9387-e8b6eb67f39d/artifact/5238c9453402ebd6/p/68a2390dec4fc115b837ead2728b68c3

File: text-inputs/src/main/java/com/decathlon/vitamin/compose/textinputs/TextInputsState.kt [hunk 1]
```diff
@@ -17,7 +21,7 @@ class TextInputStateColors(
     val focusBorderColor: Color,
     val helperColor: Color,
     val iconColor: Color,
-    @DrawableRes val icon: Int?,
+    val imageVector: ImageVector?,
```
[the remark this answers is not in the document]
Comment (GerardPaligot) [reply text-inputs/src/main/java/com/decathlon/vitamin/compose/textinputs/TextInputsState.kt:24]: Yes, but the impact is probably very limited. This icon is used to configure the warning or check icons for error and success states. We should note this breaking change in our changelog but probably nobody configure these icons in their project.

File: modals/README.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ object VitaminModals {
         onDismissRequest: () -> Unit,
         modifier: Modifier = Modifier,
         title: String? = null,
-        @DrawableRes iconRes: Int? = null,
+        painter: Painter? = null,
```
[the remark this answers is not in the document]
Comment (GerardPaligot) [reply modals/README.md:26]: No, I've forgotten to update the name of the component to `Primary2`. It is done now.

File: foundation/foundation-icons/src/main/kotlin/com/decathlon/vitamin/compose/vitaminicons/line/ChartLine.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import androidx.compose.ui.graphics.SolidColor
+import androidx.compose.ui.graphics.StrokeCap.Companion.Butt
+import androidx.compose.ui.graphics.StrokeJoin.Companion.Miter
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.graphics.vector.ImageVector.Builder
+import androidx.compose.ui.graphics.vector.path
+import androidx.compose.ui.unit.dp
+import com.decathlon.vitamin.compose.vitaminicons.LineGroup
+
+public val LineGroup.ChartLine: ImageVector
```
[the remark this answers is not in the document]
Comment (GerardPaligot) [reply foundation/foundation-icons/src/main/kotlin/com/decathlon/vitamin/compose/vitaminicons/line/ChartLine.kt:14]: I can delete it and let the automatic process add it?

[the remark this answers is not in the document]
Comment (ManonPolle) [reply foundation/foundation-icons/src/main/kotlin/com/decathlon/vitamin/compose/vitaminicons/line/ChartLine.kt:14]: We can do that yes

**Review by ManonPolle — DISMISSED** [review]

**Comment by dktbitrise** [comment]
### You can download the apk here :

**Review by ManonPolle — APPROVED** [review]

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/f9b82f2c-3fe7-46ed-859c-c5b6e4d0566c/artifact/2e681a2dc6c9b68b/p/9678a0a9c0f2c034de0c8b9344b52a11

**Pull request merged**